### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by LucaCappelletti94**
Introducing numerous new features like:

1) Support for updated Ensmallen (faster random walks and link prediction)
2) Faster computation of edge embeddings
3) Handling of non-aligned embeddings

### CODE REVIEW

**Comment by LucaCappelletti94** [comment]
If you guys agree, I can merge this and publish on Pypi.

File: embiggen/sequences/abstract_node2vec_sequence.py [hunk 1]
```diff
@@ -69,6 +65,9 @@ def __init__(
             Weight on the probability of visiting a neighbor edge of a
             different type than the previous edge. This only applies to
             multigraphs, otherwise it has no impact.
+        max_neighbours: int = None,
+            Number of maximum neighbours to consider when using approximated walks.
```
Comment (justaddcoffee) [inline embiggen/sequences/abstract_node2vec_sequence.py:69]: Possibly could add "(useful for graphs containing nodes with extremely high degrees)" - readers might not know what 'approximated walks" are

**Review by justaddcoffee — APPROVED** [review]

**Comment by justaddcoffee** [comment]
@LucaCappelletti94 looks good to me, and also works fine with HEAD of NEAT, for what that is worth.

File: embiggen/sequences/abstract_node2vec_sequence.py (hunk 1 above)
Comment (LucaCappelletti94) [reply embiggen/sequences/abstract_node2vec_sequence.py:69]: Sure! On it!

File: embiggen/embedders/glove.py [hunk 2]
```diff
@@ -37,9 +38,15 @@ def __init__(
         optimizer: Union[str, Optimizer] = "nadam",
             The optimizer to be used during the training of the model.
         alpha: float = 0.75,
-            Alpha to use for the function
+            Alpha to use for the function.
+        shared_embedding_layers: bool = False,
+            Wether to share the embedding layers weights
```
Comment (pnrobinson) [inline embiggen/embedders/glove.py:43]: wether -> whether

File: embiggen/sequences/link_prediction_sequence.py [hunk 3]
```diff
@@ -32,50 +29,44 @@ def __init__(
...
-            embedding and returns the edge embedding.
+            Can either be 'Hadamard', 'Average', 'L1', 'AbsoluteL1' or 'L2'.
         batch_size: int = 2**10,
             The batch size to use.
         negative_samples: float = 1.0,
             Factor of negatives to use in every batch.
             For example, with a batch size of 128 and negative_samples equal
             to 1.0, there will be 64 positives and 64 negatives.
+        avoid_false_negatives: bool = False,
+            Wether to filter out false negatives.
```
Comment (pnrobinson) [inline embiggen/sequences/link_prediction_sequence.py:44]: wether -> whether

File: embiggen/transformers/corpus_transformer.py [hunk 4]
```diff
@@ -84,74 +155,127 @@ def tokenize(self, texts: List[str], return_counts: bool = False, verbose: bool
...
+            new_tokens = [
+                token
+                for token in tokens
+                if token not in self._stopwords
+            ]
+            if len(new_tokens) > 0:
+                yield new_tokens
+
+    def fit(self, texts: List[str]):
         """Fit the trasformer.
```
Comment (pnrobinson) [inline embiggen/transformers/corpus_transformer.py:224]: trasformer -> transformer

**Review by pnrobinson — APPROVED** [review]
Seems fine but it is hard to assess such a big PR!
I corrected a few typos

**Pull request merged**